src/lib/checkbox/checkbox.ts

```diff
@@ -378,12 +385,15 @@ export class MatCheckbox extends _MatCheckboxMixinBase implements ControlValueAc
 
       this.toggle();
       this._transitionCheckState(
-        this._checked ? TransitionCheckState.Checked : TransitionCheckState.Unchecked);
+          this._checked ? TransitionCheckState.Checked : TransitionCheckState.Unchecked);
 
       // Emit our custom change event if the native input emitted one.
       // It is important to only emit it, if the native input triggered one, because
       // we don't want to trigger a change event, when the `checked` variable changes for example.
       this._emitChangeEvent();
+    } else if (!this.disabled && this._clickAction === 'noop') {
```

Comment:
Should this be `!== 'noop'`?

Reply:
This is because the input element's `check` is set to `true`, `indeterminate` set to `false` on user click. When the setting is `noop`, we will need to reset it to current `mat-checkbox`'s `indeterminate` and `checked`.